### PR TITLE
TimeZoneFormatter FFI and CustomTimeZone cleanup

### LINKED ISSUE #2467: Do we need TimeZoneFormatterConfig?
I think everything that `TimeZoneFormatterConfig` can do is already supported by the constellation of `load_` methods. It is mostly used internally and in tests.

I think the original idea behind `TimeZoneFormatterConfig` was to support the UTS 35 fallback logic, but it isn't actually doing that. It does nothing more than feed a `match` statement around all the `load_` functions.

Can we make `TimeZoneFormatterConfig` private for 1.0?

@nordzilla @samchen61661 @Manishearth

### PR DESCRIPTION
**Pull request by sffc**
Fixes #2467

### CODE REVIEW

**Comment by sffc** [comment]
This is a standalone reviewable PR. I am also continuing my work on the TimeZoneFormatter FFI.

**Comment by sffc** [comment]
OK, well, CI is failing, so I'll just keep pushing to this branch.

**Comment by sffc** [comment]
I added the TimeZoneFormatter FFI. This PR is ready for review

File: ffi/diplomat/src/timezone_formatter.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        Never,
+    }
+
+    impl ICU4XTimeZoneFormatter {
+        /// Creates a new [`ICU4XTimeZoneFormatter`] from locale data.
+        ///
+        /// Uses localized GMT as the fallback format.
+        #[diplomat::rust_link(icu::datetime::TimeZoneFormatter::try_new_unstable, FnInStruct)]
+        #[diplomat::rust_link(icu::datetime::time_zone::FallbackFormat, Enum, compact)]
+        pub fn try_new_with_localized_gmt_fallback(
```
Comment (Manishearth) [inline ffi/diplomat/src/timezone_formatter.rs:54]: thought: was hoping we could avoid this, but i guess this is the best way to do it

File: ffi/diplomat/src/timezone_formatter.rs [hunk 2]
```diff
@@ -96,7 +98,120 @@ pub mod ffi {
             .into()
         }
 
+        #[diplomat::rust_link(
+            icu::datetime::TimeZoneFormatter::load_generic_non_location_long,
+            FnInStruct
+        )]
+        pub fn load_generic_non_location_long(
```
Comment (Manishearth) [inline ffi/diplomat/src/timezone_formatter.rs:105]: nit: probably worth having a line of docs for these

**Review by Manishearth — APPROVED** [review]
seems like this covers all our

File: ffi/diplomat/src/timezone_formatter.rs (hunk 2 above)
Comment (sffc) [reply ffi/diplomat/src/timezone_formatter.rs:105]: will fix in my next PR

**Pull request merged**